pennylane/gradients/gradient_transform.py

```diff
@@ -149,6 +194,13 @@ def grad_method_validation(method, tape):
     return tuple(diff_methods.values())
 
 
+def gradient_analysis_and_validation(tape, method, use_graph=True, grad_fn=None, overwrite=True):
+    """Combine the execution of _gradient_analysis and _grad_method_validation."""

```

Comment:
It would be nice to have a bit more doc here as it is public facing: Something like: Update the parameter information dictionary of the tape with gradient information of each parameter. Then validates if the gradient method requested is supported by the trainable parameters of a tape, and returns the allowed parameter gradient methods.

Reply:
On that occasion, I merged the full function `_grad_method_validation`  into the new, combined function.